src/core/backbone_router/leader.hpp

```diff
@@ -68,6 +69,15 @@ class Leader : public InstanceLocator
         kStateUnchanged,      // No change on Primary Backbone Router information (only for logging).
     };
 
+    enum DomainPrefixState
+    {
+        kDomainPrefixNone = 0,  // No Domain Prefix.
```

Comment:
```suggestion
        kDomainPrefixNone = 0,  ///< No Domain Prefix.
```
Use doxygen end-of-line comments.